src/mlpack/methods/linear_svm/linear_svm_impl.hpp

```diff
@@ -54,107 +83,218 @@ LinearSVM<MatType>::LinearSVM(
+}
 
-  return out;
+template<typename ModelMatType>
+template<typename MatType,
+         typename OptimizerType,
+         typename... CallbackTypes,
+         typename, typename>
+typename LinearSVM<ModelMatType>::ElemType LinearSVM<ModelMatType>::Train(
+    const MatType& data,
+    const arma::Row<size_t>& labels,
+    const size_t numClasses,
+    OptimizerType optimizer,
+    CallbackTypes&&... callbacks)
+{
+  return Train(data, labels, numClasses, optimizer, this->lambda, this->delta,
+      this->fitIntercept, std::forward<CallbackTypes>(callbacks)...);
 }
 
-template <typename MatType>
-template <typename OptimizerType>
-double LinearSVM<MatType>::Train(
+template<typename ModelMatType>
+template<typename MatType, typename OptimizerType, typename>
+typename LinearSVM<ModelMatType>::ElemType LinearSVM<ModelMatType>::Train(
     const MatType& data,
     const arma::Row<size_t>& labels,
     const size_t numClasses,
-    OptimizerType optimizer)
+    OptimizerType optimizer,
+    const double lambda)
+{
+  return Train(data, labels, numClasses, optimizer, lambda, this->delta,
+      this->fitIntercept);
+}
+
+template<typename ModelMatType>
+template<typename MatType, typename OptimizerType, typename>
+typename LinearSVM<ModelMatType>::ElemType LinearSVM<ModelMatType>::Train(
+    const MatType& data,
+    const arma::Row<size_t>& labels,
+    const size_t numClasses,
+    OptimizerType optimizer,
+    const double lambda,
+    const double delta)
+{
+  return Train(data, labels, numClasses, optimizer, lambda, delta,
+      this->fitIntercept);
+}
+
+template<typename ModelMatType>
+template<typename MatType,
+         typename OptimizerType,
+         typename... CallbackTypes,
+         typename, typename>
+typename LinearSVM<ModelMatType>::ElemType LinearSVM<ModelMatType>::Train(
+    const MatType& data,
+    const arma::Row<size_t>& labels,
+    const size_t numClasses,
+    OptimizerType optimizer,
+    const double lambda,
+    const double delta,
+    const bool fitIntercept,
+    CallbackTypes&&... callbacks)
 {
+  this->numClasses = numClasses;
+  this->lambda = lambda;
+  this->delta = delta;
+  this->fitIntercept = fitIntercept;
+
   if (numClasses <= 1)
   {
     throw std::invalid_argument("LinearSVM dataset has 0 number of classes!");
   }
 
-  LinearSVMFunction<MatType> svm(data, labels, numClasses, lambda, delta,
-      fitIntercept);
-  if (parameters.is_empty())
+  LinearSVMFunction<MatType, ModelMatType> svm(data, labels, numClasses, lambda,
+      delta, fitIntercept);
+  const bool needNewParameters = (parameters.is_empty() ||
+      (fitIntercept && (parameters.n_rows != data.n_rows + 1)) ||
+      (!fitIntercept && (parameters.n_rows != data.n_rows)) ||
+      (parameters.n_cols != numClasses));
+  if (needNewParameters)
     parameters = svm.InitialPoint();
 
   // Train the model.
-  const double out = optimizer.Optimize(svm, parameters);
+  const double out = optimizer.Optimize(svm, parameters, callbacks...);
```

Comment:
We need to change `double` to `ElemType` in here as well, since we are returning `ElemType`

Reply:
Thanks, nice catch!  Fixed in 621ade26. :+1: